### ISSUE TITLE
Irmin pack inspector tool feedback

### ISSUE THREAD

**Issue by clecat** [mirage/irmin#1940]
This issue aims to gather all the feedback/requests that might be done for the tool developed in this [pull request](https://github.com/mirage/irmin/pull/1939) once it is merged.

Summary:
- [x] With very large files, show gets laggy as we proceed down the file, probably due to a list, fix that.
- [ ] If files already exist, take advantage of them, don't recompute (ngoguey42).
- [x] Show offsets, not just entry indices (ngoguey42)
- [x] Move to children of node and commit (ngoguey42)
- [x] Prettier print than just Repr.pp for inode and commit. (e.g. decode dict indices, show children hash, child kind) (ngoguey42) (partially done)
- [x] Move one entry at a time, not just per kind (ngoguey42)(was already done at the time)
- [x] Optimize the parsing of the file, to ensure that large files can be properly and consistently processed.
- [ ] Add a progress bar to give the user more input on what is going on (especially on longer files were in can take a minute or more)
- [x] Use the main repository for notty.
- [ ] Add on mouse movement highlighting maybe ?
- [x] Make sure that 4.14 gets supported by `notty` in order to get rid of the pin

**Comment by Ngoguey42** [comment]
Some desired QOL enhancement:
- If files already exist, take advantage of them, don't recompute.
- Show offsets, not just entry indices
- Move to children of node and commit
- Prettier print than just `Repr.pp` for inode and commit. (e.g. decode dict indices, show children hash, child kind)
- Move one entry at a time, not just per kind
